### PR TITLE
feat: optionally generate thumbnails for invalid images

### PR DESCRIPTION
**Pull request by Saschl**
First PR here, there are some questions. Came from a suggestion on discord.

### CODE REVIEW

File: server/src/repositories/media.repository.ts [hunk 1]
```diff
@@ -45,7 +46,8 @@ export class MediaRepository implements IMediaRepository {
   }
 
   async generateThumbnail(input: string | Buffer, output: string, options: ThumbnailOptions): Promise<void> {
-    const pipeline = sharp(input, { failOn: 'error', limitInputPixels: false })
+    // some invalid images can still be processed by sharp, but we want to fail on them by default to avoid crashes
+    const pipeline = sharp(input, { failOn: PROCESS_INVALID_IMAGES ? 'none' : 'error', limitInputPixels: false })
```
Comment (Saschl) [inline server/src/repositories/media.repository.ts:50]: Not really sure if using an import from constants is the right choice here. Might also include it in the `options` object, but I guess that's something that shouldn't be configurable in the UI?

File: server/src/constants.ts [hunk 2]
```diff
@@ -27,6 +27,8 @@ export const WEB_ROOT = process.env.IMMICH_WEB_ROOT || '/usr/src/app/www';
 const HOST_SERVER_PORT = process.env.IMMICH_PORT || '2283';
 export const DEFAULT_EXTERNAL_DOMAIN = 'http://localhost:' + HOST_SERVER_PORT;
 
+export const PROCESS_INVALID_IMAGES = process.env.IMMICH_PROCESS_INVALID_IMAGES === 'true';
```
Comment (Saschl) [inline server/src/constants.ts:30]: Would this need to be documented as well (independent from where the option will be in the end)?

File: server/src/repositories/media.repository.ts (hunk 1 above)
Comment (mertalev) [reply server/src/repositories/media.repository.ts:50]: For testing, I think it should either be a field in `options` or the env should be provided through a function like [here](https://github.com/immich-app/immich/blob/45cdd8a4b5cddc2af8360662a0e5d3f4d1e43b87/server/src/database.config.ts#L36).

File: server/src/constants.ts (hunk 2 above)
Comment (mertalev) [reply server/src/constants.ts:30]: Yes, it should be added [here](https://github.com/immich-app/immich/blob/main/docs/docs/install/environment-variables.md).

**Comment by alextran1502** [comment]
Do you have the invalid image to attach to the PR so we can test in the future?

File: server/src/constants.ts (hunk 2 above)
Comment (Saschl) [reply server/src/constants.ts]: Done

File: server/src/repositories/media.repository.ts (hunk 1 above)
Comment (Saschl) [reply server/src/repositories/media.repository.ts]: Done that and added a test case. Not too sure about accessing the env directly in the service, let me know if you have other suggestions :)

**Comment by Saschl** [comment]
> Do you have the invalid image to attach to the PR so we can test in the future?

Sure! https://drive.google.com/file/d/1pfCPCEJkA_d-glKWOgou3rGbHixA8AK6/view?usp=sharing

**Comment by bo0tzz** [comment]
What is the reason for making this configurable/not defaulting it to on?

**Comment by Saschl** [comment]
> What is the reason for making this configurable/not defaulting it to on?

It seems like that having it on by default could cause issues like: https://github.com/immich-app/immich/issues/10452

**Comment by alextran1502** [comment]
I tried to drop the file you posted into the instance. Of course, the thumbnail doesn't get generated, but it doesn't seem to crash the instance.

**Comment by Saschl** [comment]
> I tried to drop the file you posted into the instance. Of course, the thumbnail doesn't get generated, but it doesn't seem to crash the instance.

Exactly, and with `failOn: none` the thumbnails are being generated. The fear is only that there might be images then that could cause crashes I believe

**Comment by mertalev** [comment]
It isn't guaranteed that an invalid image crashes the server, but there's a higher risk if it's routed to a foreign loader like ImageMagick, libraw or libheif. I've had it happen for a HEIF image in the past and it went away when I changed this setting to fail on error.

**Review by mertalev — APPROVED** [review]

**Pull request merged**